src/viam/__init__.py

```diff
@@ -1,71 +1,73 @@
-import sys
-from importlib.metadata import PackageNotFoundError, version
-from uuid import uuid4
-
-from viam.gen.common.v1.common_pb2 import ResourceName as _ResourceName
-
-from .logging import getLogger as _getLogger
-
-##############
-# VERSIONING #
-##############
-try:
-    __version__ = version("viam-sdk")
-except PackageNotFoundError:
-    pass
-
-##########################################################
-# #################### TASK PREFIX ##################### #
-# ###### Used to give each task a unique identifier #### #
-# so that tasks spawned by Viam can be canceled on close #
-##########################################################
-_TASK_PREFIX = uuid4().hex
-
-
-##########################################################
-# ################### LOG EXCEPTIONS ################### #
-# Change the default exception handler to log exceptions #
-# ############### prior to raising them ################ #
-##########################################################
-
-
-def _log_exceptions(exctype, value, traceback):
-    _LOGGER = _getLogger(__name__)
-    _LOGGER.error("[ERROR] Uncaught exception", exc_info=(exctype, value, traceback))
-    sys.__excepthook__(exctype, value, traceback)
-
-
-sys.excepthook = _log_exceptions
-
-
-##################
-# MONKEY PATCHES #
-##################
-def _rname_str(self: _ResourceName) -> str:
-    return f"{self.namespace}:{self.type}:{self.subtype}/{self.name}"
-
-
-_ResourceName.__str__ = _rname_str
-
-
-def _rname_repr(self: _ResourceName) -> str:
-    return f"<viam.proto.common.ResourceName {str(self)} at {hex(id(self))}>"
-
-
-_ResourceName.__repr__ = _rname_repr
-
-
-def _rname_hash(self: _ResourceName) -> int:
-    return hash(str(self))
-
-
-_ResourceName.__hash__ = _rname_hash  # type: ignore
-
-
-def _rname_eq(self: _ResourceName, other: object) -> bool:
-    if isinstance(other, _ResourceName):
-        return self.__hash__() == other.__hash__()  # type: ignore
-    return False
-
-
-_ResourceName.__eq__ = _rname_eq  # type: ignore
+import sys
+from importlib.metadata import PackageNotFoundError, version
+from uuid import uuid4
+
+from viam.gen.common.v1.common_pb2 import ResourceName as _ResourceName
+
+from .logging import getLogger as _getLogger
+
+##############
+# VERSIONING #
+##############
+try:
+    __version__ = version("viam-sdk")
+except PackageNotFoundError:
+    pass
+
+api_version = "0.1.327"
+
```

Comment:
Only lines that actually changed. `update_api_version.py` asked for the file to be re-written, but it should be the same. I also checked on diffchecker, and it was confirmed that nothing changed.

Reply:
Doesn't seem like it does, as my RDK test worked well.

Reply:
Does this mean that this file will be completely rewritten every time API version updates? That doesn't seem great to me, I don't think we should be getting into the habit of periodically ignoring ~150 lines of diff in a non-computer-generated file. Could we figure out how to more selectively replace the value, or store the version in a smaller file that only contains that information, which is then imported by `__init__.py`?